eclipse/sw360
# feat(ui): Add, Edit, Activate, Deactivate, Synchronize User from Admin - User Page

> Add, Edit, Activate, Deactivate, Synchronize User from Admin -> User Page
    * Changed Clearing Request, Clearing Team field from free text field to user search dialog in create/edit Page
> * Which issue is this pull request belonging to and how is it solving it? (*#1242* , *#1241*)
> * Did you add or update any new dependencies that are required for your change? - No

Issues:
closes #1241 
closes #1242 

### How To Test?
   - Test Create/Edit Clearing Request for Clearing Team field
      - Condition to be able to edit CR Clearing Team - `isClearingTeam and not isClosedOrRejected` or `isRequestingUser`
   - Add, Edit, Activate, Deactivate, Synchronize User from Admin -> User Page
   - Synchronize User in both the DB, Scenarios
      - User present in Liferay DB, not present in CouchDB
      - User present in CouchDB, not present in Liferay DB
   - Deactivate
       -  If user is deactivated. User should not be able to login to 
           - sw360 from UI/Browser
           - Call REST API using API tokens, created from Preference tab
           - Call REST API using existing Bearer token/ OAuth Token. Token created before deactivating the user
           - Login to sw360-authorization server and create client details - http://localhost:8080/authorization/client-management 
           - Should not be able to refresh/create new token from pre-existing OAuth client details
           - Deactivated Users should not be listed when Search for User occurs like in Project, Component, Releases, Clearing Request - add moderators, contributors, Lead Architect, Security Responsible . etc
```
curl -X POST --user '94512*******:76cee06a-0161*************' -d 'grant_type=password&username=admin2@sw360.org&password=12345' http://localhost:8080/authorization/oauth/token -k

{
  "error" : "server_error",
  "error_description" : "403 "
}
```
   - To test on Stage or Sw360 with Single Sign-On Server
      - Set `sso.login.enabled=true` in `/etc/sw360/sw360.properties`
      - Test above mentioned steps.
      - Difference On Sw360 with Single Sign-On Server than in local
         - Password field is not present and cannot be set.
         - Try deleting a colleagues user from Liferay Control Panel, Regenerate the User from Admin -> User -> Add User.
         - Verify with your colleagues, if he is able to login, create, update, view his old projects.
> Have you implemented any additional tests?- No

### Checklist
Must:

Signed-off-by: Jaideep Palit <jaideep.palit@siemens.com>
